portal-api/src/v1/api.js

```diff
@@ -267,16 +272,23 @@ const saveUtilisationReport = async (reportData, reportPeriod, user, fileInfo) =
   }
 };
 
+/**
+* @typedef {Object} ReportDetails
+* @property {string} bankId
+* @property {import('../types/utilisation-reports').ReportPeriod} reportPeriod - a report period to filter reports by
+* @property {string[]} reportStatuses - list of report statuses to filter reports by
+*/
+
 /**
  * Gets utilisation reports for a specific bank. If a report
- * period is not provided, all reports for that bank are
+ * period or statuses are not provided, all reports for that bank are
  * returned. If a report period is provided, the report
- * submitted for that report period is returned. Returned
- * reports are ordered by year and month ascending.
- * @param {string} bankId
- * @param {import('../types/utilisation-reports').ReportPeriod} [reportPeriod]
+ * submitted for that report period is returned. If an array of report 
+ * statuses are provided, the reports returned are filtered by status.
+ * Returned reports are ordered by year and month ascending.
+ * @param {ReportDetails}
  */
-const getUtilisationReports = async (bankId, reportPeriod) => {
+const getUtilisationReports = async ({ bankId, reportPeriod, reportStatuses }) => {
```

Comment:
I think this function should be defined as follows (using typescript so that I can include type information there instead):
```typescript
const getUtilisationReports = async (bankId: string, reportFilterOpts: ReportFilterOptions) => { ... };
```
The main reason I think this is that the `bankId` parameter is always required, which makes it distinct from the other two parameters which are optional (notice the introduction of square brackets in the JSDocs typedef to highlight this). Given this is the case, I think this section of the code can be rewritten like this:
```javascript
/**
* @typedef {Object} ReportFilterOptions
* @property {import('../types/utilisation-reports').ReportPeriod} [reportPeriod]
* @property {string[]} [reportStatuses]
*/

/**
 * Gets utilisation reports for a specific bank. If a report
 * period or statuses are not provided, all reports for that bank are
 * returned. If a report period is provided, the report
 * submitted for that report period is returned. If an array of report 
 * statuses are provided, the reports returned are filtered by status.
 * Returned reports are ordered by year and month ascending.
 * @param {string} bankId
 * @param {ReportFilterOptions}
 */
const getUtilisationReports = async (bankId, { reportPeriod, reportStatuses }) => {
```
What do you think? I also think it may be worthwhile to rename the function to document that the primary thing we are filtering by is the bank id, but I also don't think it's necessary given the docstring explanation 🤷